packages/client/src/index.js

```diff
@@ -84,6 +84,12 @@ const loadBudibase = async () => {
       } else {
         dndStore.actions.reset()
       }
+    } else if (type === "request-context") {
+      const { selectedComponentInstance } = get(componentStore)
+      const context = selectedComponentInstance?.getDataContext()
+      eventStore.actions.dispatchEvent("provide-context", {
+        context: JSON.stringify(context),
```

Comment:
Any scenario where this may fail?

Reply:
I don't think so - `null`, `undefined` etc stringify just fine, in case it was nullish. The only thing that would fail would be objects with cyclic references I think, but there aren't any instances of that in our codebase.

I wonder if you could force it to fail by using a custom JS datasource that deliberately includes cyclic references. Probably worth try/catching this just in case. Good call 👍